crates/polars-expr/src/expressions/binary.rs

```diff
@@ -354,7 +354,7 @@ mod stats {
         use ChunkCompareIneq as C;
         match op {
             Operator::Eq => apply_operator_stats_eq(min_max, literal),
-            Operator::NotEq => apply_operator_stats_eq(min_max, literal),
+            Operator::NotEq => apply_operator_stats_neq(min_max, literal),
```

Comment:
This is the actual fix. The rest are all drive-by's.

Reply:
auch.